xbmc/platform/win32/filesystem/Win32SMBDirectory.cpp

```diff
@@ -12,11 +12,13 @@
 #include "PasswordManager.h"
 #include "URL.h"
 #include "Win32WSDiscovery.h"
+#include "dialogs/GUIDialogBusy.h"
```

Comment:
I am not the best one to review this but afaik adding gui elements to VFS code is a no go :/
We have the same problem with python and is a nightmare. Maybe you can add this to a GUI specific file and run it async, maybe @ksooo can provide a few points as he did the same in PVR

Reply:
The GUI is blocked in any case until the search servers complete (4-5 s). If the GUI busy dialog is not displayed, it hangs anyway.

It is not an SMB issue: the same thing happens with Zeroconf or NFS.

Reply:
This is not a functional issue, it's the fact you're including GUI dependencies on a VFS/filesystem based source file. That way the architecture is degraded even more and the path to Kodi headless (one day) gets even harder.
Maybe you can look into https://github.com/xbmc/xbmc/blob/a80d1293b0579bbed7dd7d20304ba34a20519f78/xbmc/pvr/windows/GUIWindowPVRSearch.cpp and see how it is implemented. Dialogbusy is used but triggered from a GUI specific file. Maybe you can figure out a way to do something similar, without the current includes

Reply:
The general idea is that code not part of guilib should use CApplicationMessenger to present UI.

Maybe it would be useful to add a helper method to show a spinner similar to the helpers for dialogok in xbmc/messaging/helpers

Reply:
I Understand that point. Maybe later I will investigate that path but right now I think it is out of the scope of this PR. Just wanted to be a quick fix to return the SMB browse functionality.

git reset --hard HEAD~1  for now 😢

Reply:
I don't see it as a blocking issue, it's something I started cleaning up but didn't get very far. If you can look into it in a separate PR that would be great